app/user/page.tsx

```diff
@@ -1,27 +1,49 @@
 "use client";
 import { useState, useEffect } from "react";
 
+import { AlertCircle } from "lucide-react";
 import Link from "next/link";
 import { useAccount, useReadContract } from "wagmi";
 
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
 import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
 import { Badge } from "@/components/ui/badge";
 import { Button } from "@/components/ui/button";
 import { Skeleton } from "@/components/ui/skeleton";
 
 import { abi as HatsAbi } from "@/lib/hatsAbi";
 
+type HatData = {
+  details: string;
+  maxSupply: bigint;
+  supply: bigint;
+  eligibility: string;
+  toggle: string;
+  imageUri: string;
+  numChildren: bigint;
+  mutable: boolean;
+  active: boolean;
+};
+
+type HatMetadata = {
+  name: string;
+  description: string;
+};
+
 export default function AccountPage() {
   const { address, isConnected } = useAccount();
-  const [hatId, setHatId] = useState<string | null>(null);
+  const [hatData, setHatData] = useState<HatData | null>(null);
+  const [hatMetadata, setHatMetadata] = useState<HatMetadata | null>(null);
+  const [isMetadataLoading, setIsMetadataLoading] = useState(false);
+  const [metadataError, setMetadataError] = useState<Error | null>(null);
 
-  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137"; // Hats protocol contract address
-  const hatsId = BigInt("0x0000027000020000000000000000000000000000000000000000000000000000"); // https://app.hatsprotocol.xyz/trees/11155111/624?hatId=624.2
+  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137";
+  const hatsId = BigInt("0x0000027000020000000000000000000000000000000000000000000000000000");
```

Comment:
Just a reminder that we should put this into `constants.ts` I introduced in #17 when it's merged in.